app/profile/page.tsx

```diff
@@ -52,6 +56,35 @@ export default function ProfilePage() {
     enabled: !!address,
   });
 
+  useEffect(() => {
+    (async () => {
+      if (!userPositions) return;
+      setIsLoadingTokenPrices(true);
+      const uniqueCollateralTokenAddressArray = Array.from(
+        new Set(
+          userPositions.reduce<string[]>(
+            (acc, userPosition) => [...acc, userPosition.position.collateralTokenAddress],
+            []
+          )
+        )
+      );
+
+      for (const collateralTokenAddress of uniqueCollateralTokenAddressArray) {
+        const price = await queryClient.fetchQuery({
+          queryKey: ['tokenPriceUSD', collateralTokenAddress],
+          queryFn: async () => await getTokenUSDPrice(collateralTokenAddress),
+          staleTime: Infinity,
+        });
+
+        setTokenPrices(tokenPrices => ({
+          [collateralTokenAddress]: price,
+          ...tokenPrices,
+        }));
+      }
+      setIsLoadingTokenPrices(false);
+    })();
+  }, [queryClient, userPositions]);
```

Comment:
Not sure if we should do this here or call it inside `getUserPositionsComplete` and returned it inside `UserPositionComplete`.